### PR TITLE
Update resource regexp to match terraform spec

### PR DESCRIPTION
**Pull request by sergeyzenchenko**
Update resource regexp to match terraform spec

### CODE REVIEW

File: packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const parseOutput = (str: string): DeployingResource[] => {
     if (/^Outputs:/.test(line)) { return }
     if (/^data\..*/.test(line)) { return }
 
-    const resourceMatch = line.match(/^([a-zA-Z\d_\-.]*):/)
+    const resourceMatch = line.match(/^([a-zA-Z][a-zA-Z\d_\-.]*):/)
```
Comment (jsteinich) [inline packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx:39]: The spec says the first character isn't a digit, but `-` and `_` may be allowed (could test to confirm).

**Review by jsteinich — COMMENTED** [review]
Thank you for your contribution. Can you confirm if other non-letters are allowed? Making this regex more strict may eliminate false matches, but we haven't had many cases reported so I think we should be cautious here.

**Comment by sergeyzenchenko** [comment]
I can add unit test for that function. This regexp is build according to terraform spec

**Comment by jsteinich** [comment]
I just did a quick test and `_` is allowed for the first character of a resource identifier. `-` is not.
Error message from testing: "A name must start with a letter or underscore and may contain only letters,
digits, underscores, and dashes."

**Comment by skorfmann** [comment]
> I just did a quick test and `_` is allowed for the first character of a resource identifier. `-` is not.
> Error message from testing: "A name must start with a letter or underscore and may contain only letters,
> digits, underscores, and dashes."

Would be great if you could adjust it accordingly @sergeyzenchenko

**Comment by sergeyzenchenko** [comment]
Yes! I've already found that _ should be allowed. I will push update it today

File: packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx (hunk 1 above)
Comment (skorfmann) [reply packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx]: So, like this? https://github.com/hashicorp/terraform-cdk/pull/617/commits/0e7c40d75fad6d77d563775c37112720e0e524d1

Comment (jsteinich) [reply packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx]: It actually can't start with a dash.
"A name must start with a letter or underscore and may contain only letters,
digits, underscores, and dashes."

Comment (skorfmann) [reply packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx]: Oh, thanks for clarifying :)

**Pull request merged**